src/statusBar.ts

```diff
@@ -31,15 +31,22 @@ class StatusBarClass implements vscode.Disposable {
     }
   }
 
-  public SetColor(color: string) {
+  public SetColor(background: string, foreground?: string) {
     const currentColorCustomizations = vscode.workspace
       .getConfiguration('workbench')
       .get('colorCustomizations');
-    const mergedColorCustomizations = Object.assign(currentColorCustomizations, {
-      'statusBar.background': `${color}`,
-      'statusBar.noFolderBackground': `${color}`,
-      'statusBar.debuggingBackground': `${color}`,
-    });
+
+    const newColorCustomizations = {
+      'statusBar.background': `${background}`,
+      'statusBar.noFolderBackground': `${background}`,
+      'statusBar.debuggingBackground': `${background}`,
+    };
+    if (foreground !== undefined) {
+      newColorCustomizations['statusBar.foreground'] = `${foreground}`;
+    } else if (currentColorCustomizations !== undefined && 'statusBar.foreground' in currentColorCustomizations) {
```

Comment:
Should be sufficient to only have an `else`, and run `delete currentColorCustomizations['statusBar.foreground'];` if foreground is undefined.

Reply:
That would result in `[ts] Object is possibly 'undefined'`.

Reply:
Actually, this `delete` doesn't work. As you are modifying the object, but not actually updating it with VSCode.

This should be better:
```
  public SetColor(background: string, foreground?: string) {
    const currentColorCustomizations = vscode.workspace
      .getConfiguration('workbench')
      .get('colorCustomizations');

    let colorCustomizations = Object.assign(currentColorCustomizations, {
      'statusBar.background': `${background}`,
      'statusBar.noFolderBackground': `${background}`,
      'statusBar.debuggingBackground': `${background}`,
    });

    if (foreground === undefined) {
      delete colorCustomizations['statusBar.foreground'];
    } else {
      colorCustomizations['statusBar.foreground'] = `${foreground}`;
    }

    vscode.workspace
      .getConfiguration('workbench')
      .update('colorCustomizations', colorCustomizations, true);
  }
```